### PR TITLE
Refactor modules to be more configurable

### PR DESCRIPTION
**Pull request by mohsiur**
## Why

To make the modules be more configurable, and removing hardcoded integrations

## How 

* Update `main.tf` with changes from modules
* Update changes in `modules/kubernetes-goodies`
    * To accept `cluster_name` as parameter in `conf/cluster-autoscaler-chart-values.yml`
    * To pass `cluster_name` as parameter in `modules/kubernetes-goodies/main.tf`
*  Update changes in  `modules/simple-kubernetes/` 
    * Extract `asg_tags` into `locals`
    * Make worker_groups parameters as variables
* Update changes in `modules/simple-log-es`
    * Change name of resource to be generic 
    * Pass tags as variables
* Update changes in `modules/w7b6-mysql` 
    * Pass username and password as variables
    * Pass db_name as variable
    * Pass identifier and snapshot_identifier as variables
* Update `modules/w7b6`  with changes from modules

## Test
Running `terraform plan` should not result in errors and if existing infra exists, there should be 0 infrastructure changes.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -80,16 +81,35 @@ module "simple_es" {
...
+      groups   = ["system:masters"]
+    }
+  ]
+  map_roles = [
+    {
+      rolearn  = "arn:aws:iam::<ACCOUNT_NUMBER>:role/<ROLE_NAME>"
+      username = "<ROLE_NAME>"
+      groups   = ["system:masters"]
+    },
+  ]
```
Comment (imyousuf) [inline main.tf:112]: I primarily have 2 questions:
1. Why are these needed?
1. Who will replace `<ACCOUNT_NUMBER>`, `<USER_NAME>` and `<ROLE_NAME>`?

**Comment by imyousuf** [comment]
So full stack is up and running, VPN connected, webhook broker was deployed, elasticsearch worked, logs shipped (as shown in the SS). So once the question is answered above, this PR can be merged. Will also bump the version once the PR is merged.

elasticsearch_log

File: main.tf (hunk 1 above)
Comment (mohsiur) [reply main.tf:112]: `map_users` and `map_roles` allows the user access to `kubectl` and api calls against the k8s cluster. Ideally `map_users` shouldn't be used, and instead `map_roles`. It would be needed primarily for debugging and CD using Jenkins.

As for maintenance, it should be configured in the `.tfvars` or `variables.tf` files, and updated by the user accordingly. I didn't find a `.tfvars` file in the repo hence I added them here with `<>` tags but can move it accordingly.

Comment (imyousuf) [reply main.tf:112]: I see what you mean; in that case should it not be in `variables.tf` with default values and documentation set accordingly similar to how we have the [`cert_arn`](https://github.com/imyousuf/terraform-aws-webhook-broker/blob/ce8a9d3855c063c98dd9e1a98fd71dac96982dd6/variables.tf#L11)? The `.tfvars` are for module users to create and use accordingly rather than the project containing them hence you do not see any in the repo.

**Review by imyousuf — APPROVED** [review]
